### DISCUSSION TITLE
Wizard: save data on each step

### DISCUSSION THREAD

**Discussion by sambeauprak** [filamentphp/filament#11609] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

Hi,

I want to save all data from each step when on create and update resource form.
I understand ` ->afterValidation`  but I also need to set a status on my resource (draft, active, ...)

How could I do that? Do I need this package for my needs: https://github.com/GuavaCZ/filament-drafts?

Thanks for your help,
Regards

**Comment by leandrocfe** [comment, marked as answer]
You can use `afterValidation` and get the form state from each step

```php
Step::make('Step1')
    ->schema([
    ...
    ])
    ->afterValidation(function (Component $livewire) {
        $data = $livewire->form->getState();
    })
```

**Reply by juanborras** [reply]
Running your code I got:
 local.ERROR: Undefined property: Filament\Forms\Form::$state
What I am missing?
